File: stack/stack.go
package stack

import (
	"fmt"
)

type Stack struct {
	Items []Item
}

type Item interface{}

func NewStack() Stack {
	var stack Stack
	return stack
}

func (stack *Stack) Push(data Item) {
	stack.Items = append(stack.Items, data)
}

func (stack *Stack) Pop() Item {
	if len(stack.Items) > 0 {
		ele := stack.Items[len(stack.Items)-1]
		stack.Items = stack.Items[:len(stack.Items)-1]
		return ele
	}
	return nil
}

func (stack *Stack) PrintStack() {
	for i := len(stack.Items) - 1; i >= 0; i-- {
		fmt.Print(stack.Items[i])
	}
}

File: stack/stack.py
from sys import maxsize

def createStack():
    stack = []
    return stack

def isEmpty(stack):
    if len(stack) > 0:
        return False
    return True

def push(stack, item):
    stack.append(item)
    return stack

def pop(stack):
    if isEmpty(stack):
        # print(maxsize)
        return str(-maxsize-1)
    return stack.pop()

def peek(stack):
    if isEmpty(stack):
        return str(-maxsize-1)
    return stack[len(stack) - 1]

stack = createStack() 
push(stack, str(10)) 
push(stack, str(20)) 
push(stack, str(30)) 
print(pop(stack) + " popped from stack")
print(stack)
